src/cpp/fastdds/domain/DomainParticipantImpl.cpp

```diff
@@ -1813,8 +1818,43 @@ bool DomainParticipantImpl::can_qos_be_updated(
+            !(to.wire_protocol().builtin.typelookup_config.use_client == from.wire_protocol().builtin.typelookup_config.use_client) ||
+            !(to.wire_protocol().builtin.typelookup_config.use_server == from.wire_protocol().builtin.typelookup_config.use_server) ||
+            !(to.wire_protocol().builtin.metatrafficUnicastLocatorList == from.wire_protocol().builtin.metatrafficUnicastLocatorList) ||
+            !(to.wire_protocol().builtin.metatrafficMulticastLocatorList == from.wire_protocol().builtin.metatrafficMulticastLocatorList) ||
+            !(to.wire_protocol().builtin.initialPeersList == from.wire_protocol().builtin.initialPeersList) ||
+            !(to.wire_protocol().builtin.readerHistoryMemoryPolicy == from.wire_protocol().builtin.readerHistoryMemoryPolicy) ||
+            !(to.wire_protocol().builtin.readerPayloadSize == from.wire_protocol().builtin.readerPayloadSize) ||
+            !(to.wire_protocol().builtin.writerHistoryMemoryPolicy == from.wire_protocol().builtin.writerHistoryMemoryPolicy) ||
+            !(to.wire_protocol().builtin.writerPayloadSize == from.wire_protocol().builtin.writerPayloadSize) ||
+            !(to.wire_protocol().builtin.mutation_tries == from.wire_protocol().builtin.mutation_tries) ||
+            !(to.wire_protocol().builtin.avoid_builtin_multicast == from.wire_protocol().builtin.avoid_builtin_multicast) ||
+            !(to.wire_protocol().builtin.discovery_config.discoveryProtocol == from.wire_protocol().builtin.discovery_config.discoveryProtocol) ||
+            !(to.wire_protocol().builtin.discovery_config.use_SIMPLE_EndpointDiscoveryProtocol == from.wire_protocol().builtin.discovery_config.use_SIMPLE_EndpointDiscoveryProtocol) ||
+            !(to.wire_protocol().builtin.discovery_config.use_STATIC_EndpointDiscoveryProtocol == from.wire_protocol().builtin.discovery_config.use_STATIC_EndpointDiscoveryProtocol) ||
+            !(to.wire_protocol().builtin.discovery_config.discoveryServer_client_syncperiod == from.wire_protocol().builtin.discovery_config.discoveryServer_client_syncperiod) ||
+            !(to.wire_protocol().builtin.discovery_config.m_PDPfactory == from.wire_protocol().builtin.discovery_config.m_PDPfactory) ||
+            !(to.wire_protocol().builtin.discovery_config.leaseDuration == from.wire_protocol().builtin.discovery_config.leaseDuration) ||
+            !(to.wire_protocol().builtin.discovery_config.leaseDuration_announcementperiod == from.wire_protocol().builtin.discovery_config.leaseDuration_announcementperiod) ||
+            !(to.wire_protocol().builtin.discovery_config.initial_announcements == from.wire_protocol().builtin.discovery_config.initial_announcements) ||
+            !(to.wire_protocol().builtin.discovery_config.m_simpleEDP == from.wire_protocol().builtin.discovery_config.m_simpleEDP) ||
+            !(to.wire_protocol().builtin.discovery_config.ignoreParticipantFlags == from.wire_protocol().builtin.discovery_config.ignoreParticipantFlags))))
```

Comment:
Private variable `DiscoverySettings::static_edp_xml_config_` is not checked. A getter of this member is available: `static_edp_xml_config`.

I am not sure if it is better to split the code in several wrapper methods that helps the code readability. There are several nested conditions that can be difficult to read.

Reply:
I don't think I like the idea of splitting the check into different functions that are only called once from here, since they will not be reusable and their only purpose would be to move some of this code behind a courtain.